js/main.js

```diff
@@ -14,25 +14,60 @@ $photoUrl.addEventListener('input', updateSRC);
 var $form = document.forms[0];
 function storeData(event) {
   event.preventDefault();
-  var dataEntry = {};
-  dataEntry.title = $form.elements.title.value;
-  dataEntry.pictureLink = $form.elements.photoUrl.value;
-  dataEntry.notes = $form.elements.notes.value;
-  dataEntry.entryId = data.nextEntryId;
-  data.entries.unshift(dataEntry);
-  data.nextEntryId++;
-  $form.reset();
-  $previewPhoto.src = 'images/placeholder-image-square.jpg';
-  $entriesList.setAttribute('class', 'entries-list');
-  data.view = 'entries';
-  switchViewTo('entries');
+  if (data.editing === null) {
+    var dataEntry = {};
+    dataEntry.title = $form.elements.title.value;
+    dataEntry.pictureLink = $form.elements.photoUrl.value;
+    dataEntry.notes = $form.elements.notes.value;
+    dataEntry.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(dataEntry);
+    $form.reset();
+    $previewPhoto.src = 'images/placeholder-image-square.jpg';
+    $entriesList.setAttribute('class', 'entries-list');
+    data.view = 'entries';
+    switchViewTo('entries');
+    var newDataEntry = data.entries[0];
+    prependDOM(newDataEntry);
+  } else {
+    var updatedEntry = {};
+    for (var dataEntriesIndex2 = 0; dataEntriesIndex2 < data.entries.length; dataEntriesIndex2++) {
+      if (data.editing === data.entries[dataEntriesIndex2].entryId.toString()) {
+        updatedEntry.title = $form.elements.title.value;
+        updatedEntry.pictureLink = $form.elements.photoUrl.value;
+        updatedEntry.notes = $form.elements.notes.value;
+        updatedEntry.entryId = data.editing;
+        data.entries[dataEntriesIndex2] = updatedEntry;
+        var $updatedEntry = createEntry(updatedEntry);
+        $updatedEntry.setAttribute('data-entry-id', updatedEntry.entryId);
+        for (var childElementIndex = 0; childElementIndex < $entriesUnorderedList.children.length; childElementIndex++) {
+          if ($entriesUnorderedList.children[childElementIndex].getAttribute('data-entry-id') === updatedEntry.entryId) {
+            $entriesUnorderedList.children[childElementIndex].replaceWith($updatedEntry);
+          }
+        }
+        $entriesUnorderedList.prepend($updatedEntry);
```

Comment:
This looks like it would be moving the edited entry to the top of the list, which is not the intention, you want to replace the entry in-place in the DOM entry list. This can likely just be deleted

Reply:
Sorry! I thought that after an entry was updated it was to be moved to the top of the page!